Framework/Algorithms/src/WorkspaceJoiners.cpp

```diff
@@ -56,19 +56,14 @@ WorkspaceJoiners::execWS2D(API::MatrixWorkspace_const_sptr ws1,
     auto &outSpec = output->getSpectrum(i);
     const auto &inSpec = ws1->getSpectrum(i);
 
-    // Copy X,Y,E
-    outSpec.setX(XValues);
-    outSpec.dataY() = inSpec.dataY();
-    outSpec.dataE() = inSpec.dataE();
+    outSpec.setHistogram(inSpec.histogram());
```

Comment:
Nice (have not yet tested myself). Could you add this to the PR description so the information is not lost down here.